source/include/core_sntp_client.h

```diff
@@ -0,0 +1,463 @@
+/*
+ * coreSNTP v1.0.0
+ * Copyright (C) 2021 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file core_sntp_client.h
+ * @brief API of an SNTPv4 client library that can send time requests and receive time response to/from
+ * SNTP/NTP servers. The library follows the Best Practices suggested in the the SNTPv4 specification,
+ * [RFC 4330](https://tools.ietf.org/html/rfc4330).
+ * The library can be used to run an SNTP client in a dedicated deamon task to periodically synchronize
+ * time from the Internet.
+ */
+
+#ifndef CORE_SNTP_CLIENT_H_
+#define CORE_SNTP_CLIENT_H_
+
+/* Standard include. */
+#include <stdint.h>
+#include <stdbool.h>
+
+/* Include coreSNTP Serializer header. */
+#include "core_sntp_serializer.h"
+
+/**
+ * @ingroup core_sntp_callback_types
+ * @brief Interface for user-defined function to resolve time server domain-name
+ * to an IPv4 address.
+ * The SNTP client library attempts to resolve the DNS of the time-server being
+ * used every time the @ref Sntp_SendTimeRequest API is called.
+ *
+ * @param[in] pTimeServer The time-server whose IPv4 address is to be resolved.
+ * @param[out] pIpV4Addr This should be filled with the resolve IPv4 address
+ * of @a pTimeServer.
+ *
+ * @return `true` if DNS resolution is successful; otherwise `false` to represent
+ * failure.
+ */
+typedef bool ( * SntpResolveDns_t )( const char * pServerAddr,
```

Comment:
Are we all good with using bool? I forget the conversations around this data type regarding C90 compatibility.

Reply:
Yes, we are good with using `bool` type as long as we provide a `stdbool.h` implementation just like `coreMQTT` provides [here](https://github.com/FreeRTOS/coreMQTT/blob/main/source/include/stdbool.readme).
Ideally, instead of copying those files, having a single location for those files would be good.